### DISCUSSION TITLE
VRoidSDK 0.3.0 Some models have hair that doesn't flow

### DISCUSSION THREAD

**Discussion by DavidWeng916** [pixiv/vroid-sdk-developers#127] [Q&A] [answered]
When I upgraded from VRoidSDK 0.2.0 to VRoidSDK 0.3.0, I noticed that some of the model's hairs don't float anymore but some models do, can you help me to confirm this issue?

You can use this model to test.
This model in VRoidSDK 0.3.0 when I move this model her hair doesn't float, but with VRoidSDK 0.2.0 the hair floats!
[[無料配布]ぱペコ - puppet end公式オリジナル -](https://hub.vroid.com/en/characters/6686558299753530534/models/1223833843291692178)

Use the VRoidSDK 0.2.0 the model's hair will flow.

https://github.com/pixiv/vroid-sdk-developers/assets/115971193/7b995918-c4a5-4c4f-8cdf-b1c6486f93af

Use the VRoidSDK 0.3.0 the model's hair doesn't flow.

https://github.com/pixiv/vroid-sdk-developers/assets/115971193/32e8c838-683a-4a9d-a53d-d00b895857a4

**Comment by nyamadan** [comment, marked as answer]
It seems to be caused by an upgrade of UniVRM.

VRoid SDK 0.3.0 uses UniVRM 0.112.0.
UniVRM 0.110.0 does not cause the issue, but UniVRM 0.111.0 does.

This issue is still being investigated, but the following patch has made the model hairs flow.

```diff
--- a/Assets/VRoidSDK/Plugins/VRM10/Runtime/Migration/MigrationVrmSpringBone.cs
+++ b/Assets/VRoidSDK/Plugins/VRM10/Runtime/Migration/MigrationVrmSpringBone.cs
@@ -18,7 +18,6 @@ namespace UniVRM10
             float _hitRadius;
             float _stiffness;
             int[] _colliderGroups;
-            int? _center;
 
             List<Spring> _springs = new List<Spring>();
             public IReadOnlyList<Spring> Springs => _springs;
@@ -35,12 +34,6 @@ namespace UniVRM10
                 _stiffness = vrm0BoneGroup["stiffiness"].GetSingle();
                 _colliderGroups = vrm0BoneGroup["colliderGroups"].ArrayItems().Select(z => z.GetInt32()).ToArray();
 
-                var center = vrm0BoneGroup["center"].GetInt32();
-                if (center >= 0)
-                {
-                    _center = center;
-                }
-
                 if (vrm0BoneGroup.ContainsKey("bones"))
                 {
                     foreach (var vrm0Bone in vrm0BoneGroup["bones"].ArrayItems())
@@ -57,7 +50,6 @@ namespace UniVRM10
                     Name = _comment,
                     ColliderGroups = _colliderGroups,
                     Joints = new List<SpringBoneJoint>(),
-                    Center = _center,
                 };
                 _springs.Add(spring);
                 return spring;
```

**Reply by DavidWeng916** [reply]
Thank you @nyamadan, I followed your solution and VRM's hair and clothes can float now. 🙏
